syntax = "proto3";

package nb_grpc_services;

enum StatusCode {
  STATUS_SUCCESS = 0;
  STATUS_OPERATION_NOT_SUPPORTED = 1;
  STATUS_BAD_REQUEST = 2;
  STATUS_INTERNAL_ERROR = 3;
  STATUS_INVALID_GRPC_REQUEST = 4;
  STATUS_FILE_EXISTS = 5;
  STATUS_NO_SUCH_PROCESS = 6;
  STATUS_INVALID_ACTION = 7;
  STATUS_GRPC_SERVICE_UNAVAILABLE = 8;
  STATUS_GRPC_UNAUTHORIZED = 9;
  STATUS_NOT_CONFIGURED = 10;
  STATUS_ALREADY_CONFIGURED = 11;
  STATUS_NO_SUCH_DEVICE = 12;
  STATUS_PERSISTENCY_NOT_ENABLED = 13;
}

enum FwdEngine {
  FWD_ENGINE_UNSPEC = 0;
  LINUX = 1;
  VPP = 2;
}

enum SRv6Action {
  SRV6_ACTION_UNSPEC = 0;
  END = 1;
  END_X = 2;
  END_T = 3;
  END_DX4 = 4;
  END_DX6 = 5;
  END_DX2 = 6;
  END_DT4 = 7;
  END_DT6 = 8;
  END_B6 = 9;
  END_B6_ENCAPS = 10;
  END_DT46 = 11;
}

enum EncapMode {
  ENCAP_MODE_UNSPEC = 0;
  INLINE = 1;
  ENCAP = 2;
  L2ENCAP = 3;
}
